### PR TITLE
[#9] Count the number of analyzed modules

### LINKED ISSUE #9: Count the number of analyzed modules
Also, how should we display this information in the terminal? I guess, at the beginning of the output is fine.

### PR DESCRIPTION
**Pull request by chshersh**
Resolves #9

### CODE REVIEW

File: src/Stan/Analysis.hs [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+initialAnalysis :: Analysis
+initialAnalysis = Analysis
+    { analysisModuleCount    = 0
+    , analysisLinesOfCode    = 0
+    , analysisUsedExtensions = 0
+    , analysisObservations   = ()
+    }
+
+-- QUESTION: is it better to create a newtype? Also, help with the name...
+type AnalyseM a = State Analysis a
```
Comment (vrom911) [inline src/Stan/Analysis.hs:35]: For the name, I would prefer a more specific one, like `AnalysisState`. But this then raises the question, do we need the type alias/newtype then?

File: src/Stan/Analysis.hs [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+incModuleCount :: AnalyseM ()
+incModuleCount = modify' $
+    \analysis -> analysis { analysisModuleCount = analysisModuleCount analysis + 1 }
+    -- we probably will use lenses for this kind of stuff in the future...
```
Comment (vrom911) [inline src/Stan/Analysis.hs:40]: Sounds like a nice idea! :+1:

Comment (vrom911) [reply src/Stan/Analysis.hs:40]: We can start with the provided by `relude` lens module, and see if it is enough for our needs (should be)

File: src/Stan/Analysis.hs (hunk 1 above)
Comment (chshersh) [reply src/Stan/Analysis.hs:35]: @vrom911 Good point :+1: I don't mind expanding type alias explicitly

**Pull request merged**